python/lsst/the/monster/isolate_and_transform.py

```diff
@@ -0,0 +1,149 @@
+import os
+from smatch import Matcher
+
+from lsst.pipe.tasks.isolatedStarAssociation import IsolatedStarAssociationTask
+
+from lsst.the.monster.splinecolorterms import ColortermSpline
+from lsst.the.monster.refcats import GaiaXPInfo, GaiaDR3Info, SkyMapperInfo, PS1Info, VSTInfo
+from lsst.the.monster.utils import read_stars
+
+__all__ = ["MatchAndTransform"]
+
+"""
+This Python script takes a sharded catalog and outputs a shard transformed to
+DES bandpasses.
+
+The following steps are performed:
+1. Read in the Gaia shard.
+2. Run the IsolatedStarAssociationTask on the Gaia shard.
+3. Read in each catalog.
+4. Match each catalog with the Gaia DR3 catalog.
+5. For each band, interpolate the DES magnitudes.
+6. Save the catalog shard.
+"""
+
+
+class MatchAndTransform:
+    """Match catalogs to Gaia and transform them to 'the_monster'
+       reference frame.
+
+    Parameters
+    ----------
+    htmid : `int`
+        htm level 7 id of catalog(s).
```

Comment:
This isn't listed as optional (and isn't).  But is set as a keyword.  If you want to have this as keyword only (fair!) you can do `def run(self, *, htmid, catalog_list=[]`, etc.

Also, the level doesn't matter (as long as everything is matched).  This should just be `HTM id of the catalogs.`